.github/workflows/pre-submit.yml

```diff
@@ -0,0 +1,25 @@
+name: Pre submits
+on: [pull_request, workflow_dispatch]
+
+permissions: read-all
+
+jobs:
+  pre-submit:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.3.4
+
+      - name: setup-go
+        uses: actions/setup-go@bfdd3570ce990073878bf10f6b2d79082de49492 # v2.2.0
+        with:
+          go-version: "1.18"
+
+      - name: Run tests
```

Comment:
Maybe let's also include a job for a test build for the binary? May catch some compilation bugs https://github.com/slsa-framework/slsa-github-generator/blob/main/cmd/slsa-github-generator/main.go

Reply:
see https://github.com/slsa-framework/slsa-github-generator-go/blob/main/.github/workflows/pre-submit.yml, we have an additional `go build -mod=vendor`. Maybe this is implicitly done when running `go test`..? Let's just add it to be on the safe side

Reply:
Yeah, go test ./... will tell you if a package isn't buildable even if there are no tests.

Reply:
Also, building the binary won't necessarily use all packages (only those referenced), so we would need to do `go build ./...` which `go test ./...` does anyway.

Reply:
So I think this is unnecessary since `go test` is designed to catch build errors as well and `go test ./...` will catch build and test errors in all packages, not just those imported transitively from `main.go` but feel free to veto again if you really think it's necessary.

Reply:
LGTM then. If there's a link to back up our claims, even better.